### PR TITLE
[TextInput] Call onChangeText when auto-correction updates text value

### LINKED ISSUE #2552: [TextInput] Auto correction doesn't call onChangeText after submit
Hi,
when I write to TextInupt some typo, e.g. "awrsome" and tap submit, in text input is displayed awesome but this change doesn't call method set to property onChangeText.

### PR DESCRIPTION
**Pull request by RD1991**
Fixes #2552. Using part of code @dsibiski .
Let me know if we can merge this.

### CODE REVIEW

**Comment by facebook-github-bot-6** [comment]
By analyzing the blame information on this pull request, we identified @javache and @janicduplessis to be potential reviewers.

**Comment by facebook-github-bot-7** [comment]
@RD1991 updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9852/files/4e6f6c55c205669118e3e579acfa8f0b9f36d742..1219b6fec349f19aca0dd9d704452e9fc9008644)

**Comment by facebook-github-bot-0** [comment]
@RD1991 updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9852/files/1219b6fec349f19aca0dd9d704452e9fc9008644..f62e88310824d0405723e5ced319a4a673efb5f4)

File: Libraries/Text/RCTTextFieldManager.m [hunk 1]
```diff
@@ -39,8 +39,16 @@ - (BOOL)textField:(RCTTextField *)textField shouldChangeCharactersInRange:(NSRan
   }
 
   if (textField.maxLength == nil || [string isEqualToString:@"\n"]) {  // Make sure forms can be submitted via return
+    if (textField.autocorrectionType == UITextAutocorrectionTypeYes) {
```
Comment (javache) [inline Libraries/Text/RCTTextFieldManager.m]: So what if you're using UITextAutocorrectionTypeDefault? That would still be broken. Are there no other callbacks we get after the auto correct suggestion gets committed?

Comment (RD1991) [inline Libraries/Text/RCTTextFieldManager.m]: 'Are there no other callbacks we get after the auto correct suggestion gets committed?'
I don't think so.

Comment (hramos) [inline Libraries/Text/RCTTextFieldManager.m]: What about `UITextAutocorrectionTypeDefault`, do we need additional handling for that case?

Comment (mkonicek) [inline Libraries/Text/RCTTextFieldManager.m]: Here are the possible types:
https://developer.apple.com/reference/uikit/uitextautocorrectiontype?language=objc

Would changing the line above be what we want here?

    if (textField.autocorrectionType != UITextAutocorrectionTypeNo)

Comment (mkonicek) [inline Libraries/Text/RCTTextFieldManager.m]: Some info on what "default" means:
http://stackoverflow.com/questions/24309845/when-does-uitextautocorrectiontypedefault-not-result-in-autocorrection-being-ena

Comment (mkonicek) [inline Libraries/Text/RCTTextFieldManager.m]: What I'm confused about is the name of this function is `shouldChangeCharactersInRange` and based on that I'd expect it to return a `YES` or `NO`. But here we override the text (side effect): `textField.text =`, and then return `NO`.

**Comment by lacker** [comment]
I think this pull request has been abandoned, so I am going to close it. If you are still interested in working on this then please feel free to reopen!

**Comment by adbl** [comment]
@RD1991 do you have any updated on this issue?

**Comment by shergin** [comment]
The issue was fixed in a341e9d916e73c00842b339e42385f2335e5e413 and 9a9ecea873e686287369db76359420fb15e918e8.

**Pull request closed without merging**